00_Base/src/services/credentials.service.ts

```diff
@@ -792,4 +857,92 @@ export class CredentialsService {
 
     return storedServerCredentialsRoles;
   }
+
+  private async putCredentialsToMSP(
+    versionNumber: VersionNumber,
+    credentialsDTO: CredentialsDTO,
+    clientToken: string,
+    clientCredentialsUrl: string,
+  ): Promise<CredentialsDTO> {
+    this.credentialsClientApi.baseUrl = clientCredentialsUrl;
+
+    const putCredentialsResponse =
```

Comment:
If you can please add a try catch around this external requests to apis here, but also anything with databases, interfaces, etc. anything that can fail with an unknown exception I suppose